csrc/sparse/cutlass/sparse_compressor.cu

```diff
@@ -0,0 +1,200 @@
+  using ElementD = typename Gemm::ElementD;
+
+  // Just a dummy value
+  int32_t n = 128;
+
+  int64_t lda = a.stride(0);
+
+  using StrideA = Stride<int64_t, Int<1>, int64_t>;
+  using StrideB = Stride<int64_t, Int<1>, int64_t>;
+  using StrideC = typename Gemm::StrideC;
+
+  StrideA a_stride{lda, Int<1>{}, 0};
+
+  using GemmKernel = typename Gemm::GemmKernel;
+  typename GemmKernel::ProblemShape prob_shape{m, n, k, 1};
+
+  using LayoutA = typename GemmKernel::CollectiveMainloop::LayoutA;
+  using LayoutE = typename GemmKernel::CollectiveMainloop::LayoutE;
+
+  using ElementE = typename GemmKernel::CollectiveMainloop::ElementE;
+  using SparseConfig = typename GemmKernel::CollectiveMainloop::SparseConfig;
+
+  LayoutA a_layout = SparseConfig::fill_layoutA(prob_shape);
+  LayoutE e_layout = SparseConfig::fill_layoutE(prob_shape);
+
+  // typename Gemm::GemmKernel::ProblemShape prob_shape{m, 1, k, 1};
+
+  // Offline compressor kernel
+  using CompressorUtility =
+      cutlass::transform::kernel::StructuredSparseCompressorUtility<
+          ProblemShape, ElementA, LayoutTagA, SparseConfig>;
+
+  using CompressorKernel =
+      cutlass::transform::kernel::StructuredSparseCompressor<
+          ProblemShape, ElementA, LayoutTagA, SparseConfig,
+          cutlass::arch::Sm90>;
+
+  using Compressor =
+      cutlass::transform::device::TransformUniversalAdapter<CompressorKernel>;
+
+  auto [M, N, K, L] = prob_shape;
+
+  StrideA stride_A;
+  StrideA stride_A_compressed;
+  StrideE stride_E;
+
+  stride_A =
+      cutlass::make_cute_packed_stride(StrideA{}, cute::make_shape(M, K, L));
+
+  CompressorUtility compressor_utility(prob_shape, stride_A);
+
+  int ME = compressor_utility.get_metadata_m_physical();
+  int KE = compressor_utility.get_metadata_k_physical();
+  int KC = compressor_utility.get_tensorA_k_physical();
+
+  auto a_ptr = static_cast<ElementA*>(a.data_ptr());
+
+  // cutlass::DeviceAllocation<typename Gemm::ElementA> block_A;
+  // cutlass::DeviceAllocation<typename Gemm::ElementA> block_A_compressed;
+  // cutlass::DeviceAllocation<typename Gemm::CollectiveMainloop::ElementE>
+  // block_E;
+
+  auto a_compressed_ptr = static_cast<ElementA*>(a_compressed.data_ptr());
+  auto e_ptr =
+      static_cast<typename Gemm::CollectiveMainloop::ElementE*>(e.data_ptr());
+
+  // block_A_compressed.reset(M * KC * L);
+  // block_E.reset(ME * KE * L);
+
+  stride_A_compressed =
+      cutlass::make_cute_packed_stride(StrideA{}, cute::make_shape(M, KC, L));
+  stride_E =
+      cutlass::make_cute_packed_stride(StrideE{}, cute::make_shape(ME, KE, L));
+
+  // // Random sparsification is performed on host
+  // std::vector<ElementA> block_A_host(m * k);
+  // cutlass::device_memory::copy_to_host(block_A_host.data(), a_ptr, m * k);
+  // compressor_utility.structure_sparse_zero_mask_fill(block_A_host.data(),
+  // 2024); cutlass::device_memory::copy_to_device(a_ptr, block_A_host.data(), m
+  // * k);
```

Comment:
ditto: remove uses of `make_cute_packed_stride` where possible and clean up commented out code

Reply:
This is also the original code provided by CUTLASS in the examples.